cli/scancommands.go

```diff
@@ -331,6 +331,30 @@ func AuditCmd(c *components.Context) error {
 		}
 	}
 	auditCmd.SetTechnologies(technologies)
+
+	allSubScans := utils.GetAllSupportedScans()
+	subScans := []utils.SubScanType{}
+	for _, subScan := range allSubScans {
+		if subScan == utils.ContextualAnalysisScan {
+			if c.GetBoolFlagValue(flags.NoContextualAnalysis) {
+				if !c.GetBoolFlagValue(flags.Sca) {
+					// No CA flag provided but sca flag is not provided as well, error
+					return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("flag '--%s' cannot be used without '--%s'", flags.NoContextualAnalysis, flags.Sca), c)
+				}
+				continue
+			}
+		} else if !c.GetBoolFlagValue(subScan.String()) {
```

Comment:
meaning if a user will use just "--secrets" and not "--secrets=true" - it won't work?

Reply:
it will work both ways. `--secrets` == `--secrets=true` != `--secrets=false` or not exists